subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/RegularFileSnapshot.java

```diff
@@ -31,7 +31,11 @@
     private final FileMetadata metadata;
 
     public RegularFileSnapshot(String absolutePath, String name, HashCode contentHash, FileMetadata metadata) {
-        super(absolutePath, name);
+        this(absolutePath, name, contentHash, metadata, false);
+    }
```

Comment:
Why add this convenience constructor? I think it would be great to acknowledge at each call site the fact that we accounted for whether or not the thing we have is a symlink or not.

Reply:
The convenience constructor is not there any more.